File: atv15/exerc15.py
class SaldoInsuficienteException(Exception):
    pass

class ContaBancaria:
    def __init__(self, saldo_inicial):
        self.saldo = saldo_inicial

    def sacar(self, valor):
        if valor > self.saldo:
            raise SaldoInsuficienteException("Saldo insuficiente para saque.")
        self.saldo -= valor

    def get_saldo(self):
        return self.saldo

conta = ContaBancaria(500)

try:
    conta.sacar(600)
except SaldoInsuficienteException as e:
    print(e)

File: atv15/exerc15.go
package main
import "fmt"

type SaldoInsuficienteError struct {
    Message string
}

func (e *SaldoInsuficienteError) Error() string {
    return e.Message
}

type ContaBancaria struct {
    Saldo float64
}

func (c *ContaBancaria) Sacar(valor float64) error {
    if valor > c.Saldo {
        return &SaldoInsuficienteError{Message: "Saldo insuficiente para saque"}
    }
    c.Saldo -= valor
    return nil
}

func main() {
    conta := ContaBancaria{Saldo: 500}

    err := conta.Sacar(600)
    if err != nil {
        fmt.Println(err)
    }
}
